src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -808,6 +808,46 @@ LFORTRAN_API int32_t _lpython_bit_length8(int64_t num)
     return res;
 }
 
+LFORTRAN_API int32_t _lpython_bit_count1(int8_t num)
+{
+    int32_t res = 0;
+    num = abs(num);
+    for(; num; num >>= 1){
+        res += num & 1;
+    }
+    return res;
+}
+
+LFORTRAN_API int32_t _lpython_bit_count2(int16_t num)
+{
+    int32_t res = 0;
+    num = abs(num);
+    for(; num; num >>= 1){
+        res += num & 1;
+    }
```

Comment:
This is fine but implementing it in C runtime library doesn't feel good. Its so simple that can be done directly in the ASR itself. Replace `IntegerBitCount` with the above loop using ASR APIs, so the whole implementation will be inlined in each backend and no overhead of any function call.

I am leaning towards the ASR approach for now. I don't think we need this feature urgently, so if we do it correctly then it would be better.